src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -163,6 +163,60 @@ public S filter(Predicate<? super T> predicate) {
     public <R> StreamEx<R> map(Function<? super T, ? extends R> mapper) {
         return new StreamEx<>(stream().map(mapper), context);
     }
+    /**
+     * Returns a stream consisting of the results of replacing each element of
+     * this stream with the contents of a mapped Stream produced by applying
+     * the provided mapping function to each element.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">intermediate
+     * operation</a>.
+     *
+     * <p>
+     * The {@code mapAndFlatten()} operation has the effect of applying a
+     * one-to-many transformation to the elements of the stream, and then
+     * flattening the resulting elements into a new stream.
+     *
+     * @param <R> The element type of the new stream
+     * @param mapper a <a
+     *        href="package-summary.html#NonInterference">non-interfering </a>,
+     *        <a href="package-summary.html#Statelessness">stateless</a>
+     *        function to apply to each element which produces a
+     *        stream of new values
+     * @return the new stream
+     * @since 0.7.3
+     */
+    public <R> StreamEx<R> mapAndFlatten(Function<? super T, ? extends Stream<? extends R>> mapper) {
+        return new StreamEx<>(stream().map(mapper).flatMap(Function.identity()), context);
+    }
+    /**
+     * Flattens a {@code Stream<Stream<T>>} (stream of stream) to a single {@code Stream} of {@code T}.  
+     * Each mapped stream is
+     * {@link java.util.stream.BaseStream#close() closed} after its contents
+     * have been placed into this stream.  (If a mapped stream is {@code null}
+     * an empty stream is used, instead.)
+     *
+     * <p>This is an <a href="package-summary.html#StreamOps">intermediate
+     * operation</a>.
+     *
+     * @apiNote
+     * The {@code flatMap()} operation has the effect of applying {@link Function#identity()}
+     * transformation to the elements of the stream, and then flattening the
+     * resulting elements into a new stream.
+     *
+     *
+     * @param <R> The element type of the new stream
+     * @param mapper a <a href="package-summary.html#NonInterference">non-interfering</a>,
+     *               <a href="package-summary.html#Statelessness">stateless</a>
+     *               function to apply to each element which produces a stream
+     *               of new values
+     * @return the new stream
+     * @since 0.7.3
+     */
+    @SuppressWarnings("unchecked")
+    public <R> StreamEx<R> flatMap() {
```

Comment:
This method is unsafe from the type system perspective. It will work only for Stream of Streams and will fail with ClassCastException otherwise. I don't like adding methods that violate the type safety: all type errors should be caught at compilation time. The only exception is `sorted()` method, just because it's implemented this way in the standard Java library.

In general, it's better to discuss new API ideas in issues before submitting the pull request.

Reply:
My bad, you can discard this one.

I take note on your request to discuss new API before submitting